### DISCUSSION TITLE
Near Real-time

### DISCUSSION THREAD

**Discussion by celilo** [AzuraCast/AzuraCast#5411] [Q&A] [answered]
I am sending a stream from Butt to Azuracast and it works fine except there is a delay of 13 seconds between the moment that Butt sends the stream and Azuracast ques it. If I send the same stream directly to an Icecast instance that is not managed by Azuracast, the delay can be measured in milliseconds.

Is there a parameter in Azuracast that I can change so that I can achieve the same responsiveness?

**Comment by ghost** [comment]
not sure if this will help so please don't quot me. I read in one of the post that you may need to change the buffering settings. Seem that buffering might be causing the delay you are experiencing

**Comment by eranwo** [comment, marked as answer]
@celilo 
I do not understand something here..  How do you manage to get a delay of milliseconds in icecast at all ??   Usually there is a delay of a few seconds up to 10-15 seconds minimum because of the behavior of the icecast server itself.  I would love to know how you come to these great results

**Reply by celilo** [reply]
Thanks for deflating my hypothesis:) I was using a signal generator when testing Icecast and, because the sound was constant, I didn't notice that the audio was actually cached in my browser.  I had moved further along by the time I had moved to Azuracast and was using a live source that exposed the latency.

I was able to reduce the latency by turning off burst-to-connect in my Icecast config, but achieved nowhere near real-time. Looks like I need to turn to a different alternative for my use case of streaming real-time audio to hearing impaired devices. Azuracast scheduling features would have yielded a tremendous implementation if not for the latency.

I can get the audio to my streaming server in realtime, so maybe I'll look into using FFMPEG to create my http stream.

Thanks

**Reply by BusterNeece** [reply]
@celilo We do use the Icecast-KH branch of Icecast, which supports an initial data burst to connecting listeners and can help a little bit with the latency, but unfortunately the combined latency of the full stack of software involved is something that's beyond our control.

**Reply by celilo** [reply]
I get it. No piece of software can be everything to all use cases and the demands for real-time hearing assist are very different than radio. I appreciate the work that Azuracast has done and I've learned something from it!

FYI, I successfully grabbed the audio only part from an NDI stream using FFMPEG, and sent the output to VLC. Unfortunately I have to build a web page to provide the stream since I don't want my users to need anything other than a browser. The good news is that my latency requirements have been met in my prototype.

**Reply by rija1** [reply]
@celilo 
I need to produce multiple concurrent audio streams for conference live translation in several languages. As in your case I tried Icecast first without satisfying latency.
Could you please give me more details regarding your implementation ? How did you implement the NDI stream and FFMPEG ? Thanks

**Reply by celilo** [reply]
I've long since moved on and do not have any of my code or much recollection of the process.. 

I believe that NDI was removed from FFMPEG for licensing reasons, but you may be able to compile your own version of FFMPEG with NDI. Since VLC supported, both Icecast and ND at the time of my work, it was a simple command line conversion.
 See here: https://github.com/lplassman/FFMPEG-NDI

The NDI toolkit also includes the "NDI VLC Plugin". I assume that you could use command line VLC with this plugin to achieve the same result. 

Sorry that I can't offer more specific answers.

**Reply by rija1** [reply]
@celilo That's fine, thanks for your answer!
